scripts/scil_remove_similar_streamlines.py

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""
+Remove very similar streamlines from a bundle.
+Uses clustering to speed up the process. Streamlines are considered as similar
+based on a MDF threshold within each cluster. Can be used with large bundles,
+but the clustering parameters will need to be adjusted.
+
+The algorithm still use a system of chunks to ensure the amount of comparison
+(n**2) does not grow out of control. To overcome limitations related to this
+use of chunks, multiple iterations must be done until a convergence threshold
+is acheived.
+"""
+
+
+import argparse
+from itertools import repeat, chain
+import logging
+import multiprocessing
+import random
+from time import time
+
+from dipy.io.stateful_tractogram import Space, StatefulTractogram
+from dipy.io.streamline import save_tractogram
+from nibabel.streamlines import ArraySequence
+
+from scilpy.io.streamlines import load_tractogram_with_reference
+from scilpy.segment.models import subsample_clusters
+from scilpy.io.utils import (add_overwrite_arg,
+                             add_verbose_arg,
+                             add_reference_arg,
+                             assert_inputs_exist,
+                             assert_outputs_exist)
+from dipy.segment.clustering import qbx_and_merge
+
+
+def multiprocess_subsampling(args):
+    streamlines = args[0]
+    min_distance = args[1]
+    cluster_thr = args[2]
+    min_cluster_size = args[3]
+    average_streamlines = args[4]
+
+    min_cluster_size = max(min_cluster_size, 1)
+    thresholds = [40, 30, 20, cluster_thr]
+    cluster_map = qbx_and_merge(ArraySequence(streamlines),
+                                thresholds,
+                                nb_pts=20,
+                                verbose=False)
+
+    return subsample_clusters(cluster_map, streamlines, min_distance,
+                              min_cluster_size, average_streamlines)
+
+
+def _buildArgsParser():
+    p = argparse.ArgumentParser(formatter_class=argparse.RawTextHelpFormatter,
+                                description=__doc__)
+
+    p.add_argument('in_bundle',
+                   help='Path of the input bundle.')
+
+    p.add_argument('min_distance', type=float,
+                   help='Distance threshold for 2 streamlines to be '
+                        'considered similar (mm).')
```

Comment:
Can you add a few used cases of typical distance threshold? I have no clue if 1mm or 5mm is the right kind of options. Maybe here or above in the help. 

Something like, for example, for a CST -> X mm, for a UF -> Y mm...